sql/api/src/main/scala/org/apache/spark/sql/functions.scala

```diff
@@ -1147,6 +1147,157 @@ object functions {
    */
   def sum_distinct(e: Column): Column = Column.fn("sum", isDistinct = true, e)
 
+  /**
+   * Aggregate function: returns the concatenation of non-null input values.
+   *
+   * @group agg_funcs
+   * @since 4.0.0
+   */
+  def listagg(e: Column): Column = Column.fn("listagg", e)
+
+  /**
+   * Aggregate function: returns the concatenation of non-null input values.
+   *
+   * @group agg_funcs
+   * @since 4.0.0
+   */
+  def listagg(columnName: String): Column = listagg(Column(columnName))
```

Comment:
Can we avoid having `String` signature but only `Column`?

Reply:
Yes, of course. I saw that many functions have this overloading, so I added it.
Is it something deprecated?

Reply:
Removed all functions with string